File: src/client/d-tix/src/models/movies/MovieDto.ts
import CategoryDto from "../categories/CategoryDto";
import { MovieImageDto } from "../images/MovieImageDto";
import MovieCommentDto from "@/models/comments/MovieCommentDto";

export default class MovieDto{
    id: number;
    name: string;
    description: string;
    point: number;
    time: number;
    createdDate: Date;
    updatedDate: Date;
    categoryId: number;
    category: CategoryDto;
    movieImages: MovieImageDto[];
    movieComments: MovieCommentDto[];
}

File: src/server/MicroServices/Cinema.Services.MovieAPI/Application/Dtos/Movies/MovieDto.cs
﻿using Cinema.Services.MovieAPI.Application.Dtos.Categories;
using Cinema.Services.MovieAPI.Application.Dtos.Comments;
using Cinema.Services.MovieAPI.Application.Dtos.Files;

namespace Cinema.Services.MovieAPI.Application.Dtos.Movies
{
    public class MovieDto
    {
        public MovieDto()
        {
            MovieImages = new();
        }
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double Point { get; set; }
        public double Time { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
        public int CategoryId { get; set; }
        public CategoryDto Category { get; set; }
        public List<MovieImageDto> MovieImages { get; set; }
        public List<MovieCommentDto> MovieComments { get; set; }
    }
}
